### ISSUE TITLE
ServerSideRender: store and apply previous render height while loading/refreshing

### ISSUE THREAD

**Issue by rchipka** [WordPress/gutenberg#15746]
If you have a post/page with lots of ServerSideRender blocks, the editor scroll height will jump around a lot while these blocks are loading or refreshing.

This is especially annoying when the rendered content height is actually smaller than the loading indicator height, since it causes the scroll height to expand and contract as the blocks load.

I think the user experience would be significantly improved if ServerSideRender blocks would store the rendered height before any load/refresh.

This would allow the ServerSideRender loading indicator box to take on a more appropriate size when the block is refreshing, causing no (or very minimal) scroll height jumping.

Ideally this height value would be stored upon saving the post/page, so that the editor loads as smoothly as possible the next time you edit the page.

If ServerSideRender could store their previous heights, content editors would be able to start editing blocks located near the bottom of the editor without having to first wait for every single ServerSideRender block to load.

**Comment by paaljoachim** [comment]
Is this issue still relevant?

**Comment by rchipka** [comment]
I developed and still use a custom plugin to deal with this.

**Comment by paaljoachim** [comment]
Hi @rchipka Robbie
What about creating a PR suggesting your solution to be integrated into the Gutenberg plugin?
Or suggesting your solution here and then we can have a core developer take a look.

**Comment by amir2mi** [comment]
What about preserving the result until new content? here #28289 and here dgwyer/server-side-render-x#2

![89642258-6395f280-d8ab-11ea-82b5-6cbba42ae72f](https://user-images.githubusercontent.com/64708228/123460724-50902c00-d5fd-11eb-90a3-c0f118d77cf0.gif)

**Comment by rchipka** [comment]
@amir2mi that works well after initial load, but on large pages filled with both SSR and inline blocks, a user might scroll down to the section they want to edit and the scroll position will move significantly while the user is trying to edit and SSR blocks are loading in. Storing previous height would solve this.

**Comment by amir2mi** [comment]
@rchipka Yeah, that was also what I tried at first when I started working with the SSR component but this solution still causes the white space, this solves the scrolling problem however still gives nothing.

On the other hand, the idea of  #28289 is to remove the loading scene. It just needs loading for the initial load (when Gutenberg is starting) and then, every time the old content will be there until a successful response or an error.
It is also obvious in the gif that I have attached.

**Comment by andrea-calligaris** [comment]
> Is this issue still relevant?

Still relevant in 2022.

https://user-images.githubusercontent.com/10872697/161400221-c079fd36-3ad2-410d-89a5-07590d6e0995.mp4

Notice:
* The scrollbar on the right is moving all the time because the page is changing its height all the time.
* I'm cliking all the time but I get no response from the toolbar.

**This is set as "low priority" and "enhancement", but this is a bug that is making working on a long page (not even that long) with reusable blocks pretty much impossible for an entire minute or even more.**

**Comment by paaljoachim** [comment]
I will let @gziolo @talldan know about this issue.

**Comment by andrea-calligaris** [comment]
I should also point out that loading the page by hitting moderately fast the "Page down" key, results in loading the page rather fast and smoothly. So the issue seems to be more something about jumping in the middle of the page as demonstrated in the video, then the loading gets messy.

**Comment by Mamaduka** [comment]
Most of it has been resolved by https://github.com/WordPress/gutenberg/pull/28289.

The previously rendered content is rendered while updated content is fetched. This doesn't/can't work on initial render, since there's no previous content.

Closing as fixed.
